compiler/rustc_codegen_ssa/src/debuginfo/type_names.rs

```diff
@@ -94,7 +94,12 @@ pub fn push_debuginfo_type_name<'tcx>(
             push_debuginfo_type_name(tcx, inner_type, true, output, visited);
 
             if cpp_like_names {
-                output.push('*');
+                match *inner_type.kind() {
+                    ty::Slice(_) | ty::Str => {}
```

Comment:
Could you add a comment here explaining why `Slice` and `Str` are special-cased? (This can essentially be taken from the PR description.)